x/dymns/types/string_list.go

```diff
@@ -0,0 +1,73 @@
+package types
+
+import "sort"
+
+// StringList is a list of strings.
+// Used to add some operations on the list.
+type StringList []string
+
+// Distinct returns a new list with duplicates removed.
+// Result will be sorted.
+func (m StringList) Distinct() (distinct StringList) {
+	uniqueElements := make(map[string]bool)
+	// Describe usage of Go Map: used to store unique elements, later result will be sorted
+	defer func() {
+		distinct.Sort()
+	}()
+
+	for _, ele := range m {
+		uniqueElements[ele] = true
+	}
+
+	distinctElements := make([]string, 0, len(uniqueElements))
+	for name := range uniqueElements {
+		distinctElements = append(distinctElements, name)
+	}
+
+	distinct = distinctElements
+
+	return
+}
```

Comment:
if it's already sorted you can use slices.compact

Reply:
discussed, ignore